packages/governance/src/governor/extensions/governor_votes_quorum_fraction.cairo

```diff
@@ -0,0 +1,212 @@
+        +Drop<TContractState>
+    > of GovernorComponent::GovernorVotesTrait<TContractState> {
+        /// See `GovernorComponent::GovernorVotesTrait::clock`.
+        fn clock(self: @GovernorComponentState<TContractState>) -> u64 {
+            // VotesComponent uses the block timestamp for tracking checkpoints.
+            // That must be updated in order to allow for more flexible clock modes.
+            starknet::get_block_timestamp()
+        }
+
+        /// See `GovernorComponent::GovernorVotesTrait::CLOCK_MODE`.
+        fn clock_mode(self: @GovernorComponentState<TContractState>) -> ByteArray {
+            "mode=timestamp&from=starknet::SN_MAIN"
+        }
+
+        /// See `GovernorComponent::GovernorVotesTrait::get_votes`.
+        fn get_votes(
+            self: @GovernorComponentState<TContractState>,
+            account: ContractAddress,
+            timepoint: u64,
+            params: Span<felt252>
+        ) -> u256 {
+            let contract = self.get_contract();
+            let this_component = GovernorVotesQuorumFraction::get_component(contract);
+
+            let token = this_component.Governor_token.read();
+            let votes_dispatcher = IVotesDispatcher { contract_address: token };
+
+            votes_dispatcher.get_past_votes(account, timepoint)
+        }
+    }
+
+    //
+    // External
+    //
+
+    #[embeddable_as(QuorumFractionImpl)]
+    impl QuorumFraction<
+        TContractState, +HasComponent<TContractState>, +Drop<TContractState>
+    > of IQuorumFraction<ComponentState<TContractState>> {
+        /// Returns the token that voting power is sourced from.
+        fn token(self: @ComponentState<TContractState>) -> ContractAddress {
+            self.Governor_token.read()
+        }
+
+        /// Returns the current quorum numerator.
+        fn current_quorum_numerator(self: @ComponentState<TContractState>) -> u256 {
+            self.Governor_quorum_numerator_history.deref().latest()
+        }
+
+        /// Returns the quorum numerator at a specific timepoint.
+        fn quorum_numerator(self: @ComponentState<TContractState>, timepoint: u64) -> u256 {
+            // Optimistic search: check the latest checkpoint.
+            // The initializer call ensures that there is at least one checkpoint in the history.
+            //
+            // NOTE: This optimization is especially helpful when the supply is not updated often.
+            let (_, key, value) = self
+                .Governor_quorum_numerator_history
+                .deref()
+                .latest_checkpoint();
+
+            if key <= timepoint {
+                return value;
+            }
+
+            // Fallback to the binary search
+            self.Governor_quorum_numerator_history.deref().upper_lookup(timepoint)
+        }
+
+        /// Returns the quorum denominator.
+        fn quorum_denominator(self: @ComponentState<TContractState>) -> u256 {
+            100
```

Comment:
It could make sense to move the magic value to constants and employ `ImmutableConfig` to make it configurable

Reply:
Good point, let's increase the precision by using 1000 instead, but let's keep it as a not configurable constant for now for the sake of simplicity, since this value probably won't be updated.